### PR TITLE
Fix incorrect broadcast address

### PR DESCRIPTION
**Pull request by faisal-shah**
On at least some linux systems, the broadcast address is unset (0.0.0.0). Firstly, this is incorrect. Secondly it is not allowed per RFC1122 3.2.1.3 (a):
https://www.rfc-editor.org/rfc/rfc1122#section-3

The broadcast address is computed from the ip address, and net mask. broadcast = ip | ~net_mask

This remains the correct vale of the broadcast address regardless of what SIOCGIFBRDADDR returns. As it will be garbage if no one ever set it (with SIOCSIFBRDADDR perhaps).

This issue was exposed when attempting to use the loopback network interface. I've tried a few different distros, and on all of them the broadcast mask was unset (0.0.0.0). However, on all of them the following route was set (127.255.255.255 being the broadcast address for 127.0.0.1/8) and resulted in broadcasts being transmitted and received properly:

$ ip route get 127.255.255.255
broadcast 127.255.255.255 dev lo table local src 127.0.0.1 uid 1000
    cache <local,brd>

### CODE REVIEW

File: socketcand.c [hunk 1]
```diff
@@ -680,23 +680,26 @@ void determine_adress() {
...
 	saddr.sin_family = AF_INET;
 	saddr.sin_addr = ((struct sockaddr_in *) &ifr.ifr_addr)->sin_addr;
 	saddr.sin_port = htons(port);
 
-	/* set broadcast adress */
+	/* calculate and set broadcast adress */
 	broadcast_addr.sin_family = AF_INET;
-	broadcast_addr.sin_addr = ((struct sockaddr_in *) &ifr_brd.ifr_broadaddr)->sin_addr;
+	broadcast_addr.sin_addr = ((struct sockaddr_in *) &ifr.ifr_addr)->sin_addr;
+    broadcast_addr.sin_addr.s_addr |= ~((struct sockaddr_in *) &ifr_mask.ifr_netmask)->sin_addr.s_addr;
```
Comment (marckleinebudde) [inline socketcand.c:700]: Nitpick: please indent with <TAB> here, too

**Review by marckleinebudde — CHANGES_REQUESTED** [review]

File: socketcand.c (hunk 1 above)
Comment (faisal-shah) [reply socketcand.c:700]: Fixed. Amended the commit and force pushed. Thank you.

Comment (marckleinebudde) [reply socketcand.c:700]: I think something went wrong.....please recheck.

Comment (marckleinebudde) [reply socketcand.c]: looks good

**Comment by marckleinebudde** [comment]
Although the style of the code is fixes, I cannot say anything to the contents of this PR. Maybe @hartkopp can help.

File: socketcand.c (hunk 1 above)
Comment (faisal-shah) [reply socketcand.c]: Can you approve the review request? It still shows up at 1 change requested.

**Comment by hartkopp** [comment]
> Although the style of the code is fixes, I cannot say anything to the contents of this PR. Maybe @hartkopp can help.

Not really! I only show up in that code section due to style fixes. I assume @dschanoeh knows better ...

**Comment by dschanoeh** [comment]
I'm not sure if I'm the right one to judge since I haven't touched or used this in years but my interpretation is that there is something wrong in those distros - I couldn't find anything regarding a deprecation of `SIOCGIFBRDADDR` or known limitations. So the original code doesn't seem obviously wrong.
Nevertheless, calculating it manually seems like a good workaround and I don't see other problems arising from this.

**Review by marckleinebudde — APPROVED** [review]

**Comment by marckleinebudde** [comment]
Ok - then let's merge this.

**Comment by faisal-shah** [comment]
Thanks for being responsive!

**Pull request merged**